app_check/integration_test/src/integration_test.cc

```diff
@@ -967,15 +973,23 @@ TEST_F(FirebaseAppCheckTest, TestFirestoreTransaction) {
           {"val", firebase::firestore::FieldValue::String("start")}}),
       "document.Set 0");
 
-  std::vector<firebase::firestore::MapFieldValue> document_snapshots;
+  struct ListenerSnapshots {
+    std::mutex mutex;
+    std::vector<firebase::firestore::MapFieldValue> snapshots;
+  };
+  auto listener_snapshots = std::make_shared<ListenerSnapshots>();
   firebase::firestore::ListenerRegistration registration =
       document.AddSnapshotListener(
-          [&](const firebase::firestore::DocumentSnapshot& result,
+          [&listener_snapshots](
```

Comment:
nit: Capture `listener_snapshots` by value, instead of by reference. Otherwise, you forego the benefits of shared_ptr.

Reply:
Doh, good point